### PR TITLE
`klangbecken` package and dynamic player support

### PR DESCRIPTION
**Pull request by smlz**
Split up the code into logical components
- `api.py`: All Rest-API related code
- `api_utils.py`: API support code
- `playlist.py`: playlist handling with analyzers and processors
- `player.py`: LiquidsoapClient to query dynamic player information
- `cli.py`: Code for all the command line commands
- `settings.py`: Global settings (playlist names, supported metadata keys, ...)

Also, lots of documentation updates plus 100% test coverage

### CODE REVIEW

**Comment by smlz** [comment]
Note: only the last commit is relevant. All the rest comes from #86.

**Comment by spameier** [comment]
This will make development easier. Looks very good!

**Comment by spameier** [comment]
One very minor thing:
```
export KLANGBECKEN_PATH="./klangbecken.py"
```
should be changed to (I think):
```
export KLANGBECKEN_PATH="python -m klangbecken"
```

**Review by spameier — APPROVED** [review]

**Comment by spameier** [comment]
Another very minor thing with black and isort (disregard if you don't mind):
- black:
```
diff --git a/klangbecken/api.py b/klangbecken/api.py
index 9ad4d1d..5b0331d 100644
--- a/klangbecken/api.py
+++ b/klangbecken/api.py
@@ -337,10 +337,7 @@ class StandaloneWebApplication:
             msg = b"Welcome to the Klangbecken API!\n"
             start_response(
                 "200 OK",
-                [
-                    ("Content-Type", "text/plain"),
-                    ("Content-Length", str(len(msg))),
-                ],
+                [("Content-Type", "text/plain"), ("Content-Length", str(len(msg))),],
             )
             return [msg]

diff --git a/setup.py b/setup.py
index f10f361..e29b2ef 100644
--- a/setup.py
+++ b/setup.py
@@ -26,9 +26,6 @@ setup(
         "Topic :: Internet :: WWW/HTTP :: WSGI :: Application",
     ],
     install_requires=["docopt", "mutagen", "PyJWT >= 2.0.0", "Werkzeug"],
-    extras_require={
-        "dev": ["tox", "black", "isort"],
-        "test": ["flake8", "coverage"],
-    },
+    extras_require={"dev": ["tox", "black", "isort"], "test": ["flake8", "coverage"],},
     entry_points={"console_scripts": ["klangbecken=klangbecken.cli:main"]},
 )
```
- isort
```
diff --git a/setup.py b/setup.py
index f10f361..edf8c2f 100644
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-from setuptools import setup, find_packages
+from setuptools import find_packages, setup

 setup(
     name="klangbecken",
```

**Comment by smlz** [comment]
@spameier Thanks a lot for the feedback. The things should be fixed with the new commits.

Sorry, lots of changes again ... :sweat:

**Comment by smlz** [comment]
My changes messed up the `check_next_func` in our liquidsoap script. 5a671c1 fixes it.

File: klangbecken/cli.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+
+        if files:
+            err("ERROR: Dangling files:", ", ".join(files))
+        if playlist_counts:
+            err("ERROR: Dangling playlist entry:", ", ".join(playlist_counts.keys()))
+
+    sys.exit(1 if err.count else 0)
+
+
+def playlog_cmd(data_dir, filename):
```
Comment (spameier) [inline klangbecken/cli.py:252]: I think the file `data/log/current.json` should also be handled here.

Comment (spameier) [reply klangbecken/cli.py:252]: Nevermind, I just saw that you expose this information via the API and it's already implemented in radiorabe/klangbecken-ui#116.

**Pull request merged**